charts/opentelemetry-kube-stack/templates/collector.yaml

```diff
@@ -180,11 +184,12 @@ spec:
   {{- toYaml . | nindent 4 }}
   {{- end }}
   volumes:
+  {{- if or ($collector.presets.logsCollection.enabled) ($collector.presets.logsCollection.storeCheckpoints) ($collector.presets.hostMetrics.enabled) ($collector.volumes) }}
```

Comment:
I think this needs to be placed above `volumes` so that we only add the section if there is something that will be mounted.